### PR TITLE
23358 - Fixed NR phone input

### PR DESCRIPTION
**Pull request by JazzarKarim**
Issue #: /https://github.com/bcgov/entity/issues/23358

Changes:
- Fixed the NR phone input to match what's in the current site

BRD currently in DEV:
old brd

New one here:
new brd

### CODE REVIEW

**Comment by JazzarKarim** [comment]
The [pnpm-lock.yaml](https://github.com/bcgov/business-ui/pull/28/files#diff-b47e3970b4f4887659caf2b2c45bf20637f908e6abf9f6aa6055a4c131d6aa25) file changed drastically after I did a `pnpm install` on my side. I'm guessing that's OK, right?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-28-eng2le6x.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-28-eng2le6x.web.app

**Comment by severinbeauvais** [comment]
> The pnpm-lock.yaml file changed drastically after I did a `pnpm install` on my side. I'm guessing that's OK, right?

This is why:

image

There are 2 choices:
- match the existing version
- go with the new version and make sure everything works; I'd also say, make sure this project is "compatible" with other pnpm projects

Refs:
https://github.com/bcgov/business-dashboard-ui/blob/main/pnpm-lock.yaml
https://github.com/bcgov/namerequest/blob/main/app/pnpm-lock.yaml

I don't enough to make a decision. Ask Thor...?

File: business-registry-dashboard/app/components/Form/AddNameRequest/Base.vue [hunk 1]
```diff
@@ -48,7 +42,7 @@ const validate = (state: NRSchema): FormError[] => {
   alertText.value = ''
 
   const errors = []
-  const phoneValid = z.string().regex(/^\d{3}-\d{3}-\d{4}$/).safeParse(state.phone).success
+  const phoneValid = state.phone.length <= 12
```
Comment (severinbeauvais) [inline business-registry-dashboard/app/components/Form/AddNameRequest/Base.vue:45]: How are you stripping out the '-' per the ticket description?

Comment (JazzarKarim) [reply business-registry-dashboard/app/components/Form/AddNameRequest/Base.vue:45]: Great observation! It's actually on line 84 (the logic was already there).

**Comment by JazzarKarim** [comment]
> > The pnpm-lock.yaml file changed drastically after I did a `pnpm install` on my side. I'm guessing that's OK, right?
> 
> This is why:
> 
> image
> 
> There are 2 choices:
> 
> * match the existing version
> * go with the new version and make sure everything works; I'd also say, make sure this project is "compatible" with other pnpm projects
> 
> Refs: https://github.com/bcgov/business-dashboard-ui/blob/main/pnpm-lock.yaml https://github.com/bcgov/namerequest/blob/main/app/pnpm-lock.yaml
> 
> I don't know enough to make a decision. Ask Thor...?

Great points. Sev, I spoke to Dietrich about this, he told me it should be totally fine. I can proceed with the lock file change. Worst case I can roll back. So I'm going to leave this as is for now.

File: business-registry-dashboard/app/components/Form/AddNameRequest/Base.vue (hunk 1 above)
Comment (severinbeauvais) [reply business-registry-dashboard/app/components/Form/AddNameRequest/Base.vue:45]: This comment is resolved. (I don't have permission to do that myself.)

**Review by severinbeauvais — APPROVED** [review]
👍

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**